src/tools/clippy/clippy_lints/src/transmute.rs

```diff
@@ -601,7 +628,25 @@ impl<'tcx> LateLintPass<'tcx> for Transmute {
                             );
                         }
                     },
-                    _ => return,
+                    (_, _) if can_be_expressed_as_pointer_cast(cx, e, from_ty, to_ty) => span_lint_and_then(
+                        cx,
+                        TRANSMUTES_EXPRESSIBLE_AS_PTR_CASTS,
+                        e.span,
+                        &format!(
+                            "transmute from `{}` to `{}` which could be expressed as a pointer cast instead",
+                            from_ty,
+                            to_ty
+                        ),
+                        |diag| {
+                            if let Some(arg) = sugg::Sugg::hir_opt(cx, &args[0]) {
+                                let sugg = arg.as_ty(&to_ty.to_string()).to_string();
+                                diag.span_suggestion(e.span, "try", sugg, Applicability::Unspecified);
```

Comment:
```suggestion
                                diag.span_suggestion(e.span, "try", sugg, Applicability::MachineApplicable);
```

I think this is `MachineApplicable`. The only reason everything in the `transmute` module is `Unspecified`, is that I was to lazy to check the `Applicability`, when this argument was introduced back then.

Or are there any concerns this suggestion is `MaybeIncorrect`?

Reply:
All the suggestions I have seen from this lint have been correct, as far as my understanding goes.